pkg/rhos/ocpgwdeployer.go

```diff
@@ -200,7 +203,15 @@ func (d *ocpGatewayDeployer) deployGWNode(gatewayCount int, groupName string,
 		gatewayNodesToDeploy := gatewayCount - numGatewayNodes
 
 		if d.dedicatedGWNode {
-			err = d.deployDedicatedGWNode(gatewayNodesToDeploy, status)
+			groupName := d.InfraID + internalSecurityGroupSuffix
+
+			isFound, errSG := checkIfSecurityGroupPresent(groupName, computeClient)
+
+			if errSG != nil {
+				return errSG
+			}
+
+			err = d.deployDedicatedGWNode(gatewayNodesToDeploy, isFound, status)
```

Comment:
Given the description of the change, I find this confusing: we check whether the SG is already present, and if it is, we set a flag called `addInternalSG` to `true`. That suggests that if we find the SG, we add it. Is that really the case, or are we asking `deployDedicatedGWNode` to _use_ the SG we’ve found?

Reply:
it is the former,  when it is ovn Kubernetes the internalSG does not exist. so We want to add it only if we find one.

Reply:
So we’re looking for a security group, and if we find one, we add it? Doesn’t it already exist if we find it? Or are there two different security groups?

Reply:
The internal-sg does not exist always. It is created by openPort api called during cloud prepare. For ovn Kubernetes deployment, this SG is not created

Here we are adding it to machineset only if the SG exists else we skip it.

Reply:
Yeah,  'addInternalSG' is a bit misleading, maybe it will be better to use 'useSubmarinerInternalSG' instead of  'addInternalSG' ?

Reply:
Done.